### PR TITLE
Add option to update models rather than reinstall

### PR DESCRIPTION
**Pull request by rrgeorge**
This will change the behavior of `--update-models` to only update models if there's a newer version available (or install any missing models)
Adds the option `--install-models` to force (re) installation of all models
Also adds `--update` to `scripts/install_models.py` to only update models if there's a newer version available (or install any missing models)

Will solve #502

### CODE REVIEW

File: libretranslate/init.py [hunk 1]
```diff
@@ -44,10 +47,24 @@ def check_and_install_models(force=False, load_only_lang_codes=None):
...
-            available_package.install()
+            update = False
+            if not force:
+                for pack in installed_packages:
+                    if (
+                            pack.from_code == available_package.from_code
+                            and pack.to_code == available_package.to_code
+                        ):
+                        update = True
+                        if pack.package_version < available_package.package_version:
```
Comment (pierotofy) [inline libretranslate/init.py:58]: Semantic versioning cannot be compared by string comparison:

e.g. "12.0" < "2.9" --> True (should be False)

**Comment by pierotofy** [comment]
Nice! Thanks for the PR @rrgeorge . :pray:  Just two things before this is merged:

 * Semantic versioning should be compared using component-wise logic
 * Can we rename/change the new flag to something like:

`--update-models` --> Only update if a new version is available, or a package is not installed
`--force-update-models` --> Redownload/reinstall everything

**Comment by rrgeorge** [comment]
> Nice! Thanks for the PR @rrgeorge . 🙏 Just two things before this is merged:
> 
> * Semantic versioning should be compared using component-wise logic

That's a good point. I'll fix that. And it looks like I'll need to port that change over to argos-translate too. https://github.com/argosopentech/argos-translate/blob/4a078eebe1d64d3ea673694fd4bdbd912fd00b8e/argostranslate/package.py#L203

> * Can we rename/change the new flag to something like:
> 
> `--update-models` --> Only update if a new version is available, or a package is not installed `--force-update-models` --> Redownload/reinstall everything

No problem.

**Comment by pierotofy** [comment]
Thanks!!

**Comment by PJ-Finlay** [comment]
> That's a good point. I'll fix that. And it looks like I'll need to port that change over to argos-translate too.

Good find, I fixed this in Argos Translate 1.8.2.

**Pull request merged**